std/range/package.d

```diff
@@ -6730,6 +6734,384 @@ if (isForwardRange!Source && hasLength!Source)
+        $(D hasSlicing!Source) and $(D hasLength!Source) are $(D true).
+         */
+        @property auto back()
+        {
+            import std.algorithm : max;
+            assert(!empty, "back called on empty pairs");
+            immutable len = _source.length;
+            immutable start = max(0, len - _windowSize);
+            return _source[start .. $];
+        }
+
+        /// Ditto
+        void popBack()
+        {
+            assert(!empty, "popBack() called on empty pairs");
+            immutable end = (_source.length - 1) / _windowSize * _windowSize;
+            _source.popBack;
+            // update our "future" range too
+            _endSource.popBack;
+        }
+    }
+
+private:
+    Source _source, _endSource;
+    size_t _windowSize;
+}
+
+/// Ditto
+Slides!Source slides(Source)(Source source, size_t windowSize = 2)
+if (isForwardRange!Source)
+{
+    return typeof(return)(source, windowSize);
+}
+
+///
+unittest
+{
+    import std.array: array;
+    import std.algorithm: each, equal;
+    assert([0,1,2].slides.array == [[0, 1], [1,2]]);
+    assert(iota(5).slides(2).front.equal([0,1]));
+
+    // count k-mers
+    int[dstring] d;
+    "AGAGA"d.slides.each!(a => d[a]++);
+    assert(d == ["AG"d: 2, "GA"d: 2]);
+}
+
+unittest
+{
+    import std.array: array;
+    import std.algorithm: each, equal, map;
+
+    // different window sizes
+    assert([0,1,2,3].slides(1).array == [[0], [1], [2], [3]]);
+    assert([0,1,2,3].slides(2).array == [[0, 1], [1, 2], [2, 3]]);
+    assert([0,1,2,3].slides(3).array == [[0, 1, 2], [1, 2, 3]]);
+    assert([0,1,2,3].slides(4).array == [[0, 1, 2, 3]]);
+    assert([0,1,2,3].slides(5).array == [[0, 1, 2, 3]]);
+
+    // range without index
+    assert(iota(2).slides(2).front.equal([0,1]));
+    assert(iota(3).slides(2).map!"a.array".array == [[0,1],[1,2]]);
+
+    // empty input
+    int[] d;
+    assert(d.slides.empty);
+
+    // is copyable?
+    auto e = iota(5).slides;
+    e.popFront;
+    assert(e.save.map!"a.array".array == [[1,2], [2,3],[3,4]]);
+    assert(e.map!"a.array".array == [[1,2], [2,3],[3,4]]);
+
+    // have more fun with string counting
+    int[dstring] f;
+    "AGAGA"d.slides(3).each!(a => f[a]++);
+    assert(f == ["AGA"d: 2, "GAG"d: 1]);
+}
+
+unittest
+{
+    // test slides2 step by step
+    import std.array: array;
+    auto a = [0,1,2,3].slides;
+    assert(a.front == [0,1]);
+    a.popFront();
+    assert(a.front == [1,2]);
+    a.popFront();
+    assert(a.front == [2,3]);
+    a.popFront();
+    assert(a.empty);
+}
+
+unittest
+{
+    // test slides3 step by step
+    auto a = [0,1,2,3].slides(3);
+    assert(a.front == [0,1,2]);
+    a.popFront();
+    assert(a.front == [1,2,3]);
+    a.popFront();
+    assert(a.empty);
+}
+
+unittest
+{
+    import std.array: array;
+    import std.algorithm: equal, map;
+    import std.conv: to;
+    assert(iota(3).slides(3).array.to!string == "[[0, 1, 2]]");
```

Comment:
I didn't know that its as easy as using `equal!equal([[0,1,2]]))` - very nice :)